doc/devel/development.md

```diff
@@ -0,0 +1,1932 @@
+    -   `json` - a supplied JSON object as the request body
+    -   `body` - any ASCII text as the request body. It takes precedence over `json` if both are provided at the same
+        time (not recommended).
+    -   `url` - the request URL
+    -   `uri` - alternative alias for the request URL.
+-   callback: The callback currently returns an `error` Object, the `response` and `body`. The `body` is parsed to a
+    JSON object if the `responseType` is JSON.
+
+#### Generic middlewares
+
+This collection of utility middlewares is aimed to be used to north of the IoT Agent Library, as well as in other
+HTTP-based APIs of the IoT Agents. All the middlewares follow the Express convention of `(req, res, next)` objects, so
+this information will not be repeated in the descriptions for the middleware functions. All the middlewares can be added
+to the servers using the standard Express mechanisms.
+
+##### iotagentLib.middlewares.handleError()
+
+###### Signature
+
+```javascript
+function handleError(error, req, res, next)
+```
+
+###### Description
+
+Express middleware for handling errors in the IoTAs. It extracts the code information to return from the error itself
+returning 500 when no error code has been found.
+
+##### iotagentLib.middlewares.traceRequest()
+
+###### Signature
+
+```javascript
+function traceRequest(req, res, next)
+```
+
+###### Description
+
+Express middleware for tracing the complete request arriving to the IoTA in debug mode.
+
+##### iotagentLib.middlewares.changeLogLevel()
+
+###### Signature
+
+```javascript
+function changeLogLevel(req, res, next)
+```
+
+###### Description
+
+Changes the log level to the one specified in the request.
+
+##### iotagentLib.middlewares.ensureType()
+
+###### Signature
+
+```javascript
+function ensureType(req, res, next)
+```
+
+###### Description
+
+Ensures the request type is one of the supported ones.
+
+##### iotagentLib.middlewares.validateJson()
+
+###### Signature
+
+```javascript
+function validateJson(template)
+```
+
+###### Description
+
+Generates a Middleware that validates incoming requests based on the JSON Schema template passed as a parameter.
+
+Returns an Express middleware used in request validation with the given template.
+
+###### Params
+
+-   _template_: JSON Schema template to validate the request.
+
+##### iotagentLib.middlewares.retrieveVersion()
+
+###### Signature
+
+```javascript
+function retrieveVersion(req, res, next)
+```
+
+###### Description
+
+Middleware that returns all the IoTA information stored in the module.
+
+##### iotagentLib.middlewares.setIotaInformation()
+
+###### Signature
+
+```javascript
+function setIotaInformation(newIoTAInfo)
+```
+
+###### Description
+
+Stores the information about the IoTAgent for further use in the `retrieveVersion()` middleware.
+
+###### Params
+
+-   _newIoTAInfo_: Object containing all the IoTA Information.
+
+## DB Models (from API document)
```

Comment:
Are they aligned with last version of api.md?

In my opinion, tables should be simplified to avoid duplication. The Description field is already provided in api.md. Ideally, only two columns: payload and DB field, so clearly show the mapping between them (plus a "Notes" column to be used is some additional note about the mapping is worth to be said).

Reply:
Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1371/commits/aef1589d80299764adcf9de2491063cf1aad61ad

Reply:
Added warning note here: https://github.com/telefonicaid/iotagent-node-lib/pull/1371/commits/939051140e573734ae7178a49161afe5e06dd96e and noted here: https://github.com/telefonicaid/iotagent-node-lib/issues/1329#issue-1643605738

NTC